src/cloud-api-adaptor/podvm/files/usr/local/bin/setup-scratch-storage

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -o errtrace
+
+[ -n "${DEBUG:-}" ] && set -o xtrace
+
+handle_error() {
+    local exit_code="${?}"
+    local line_number="${1:-}"
+    echo "error:"
+    echo "Failed at $line_number: ${BASH_COMMAND}"
+    exit "${exit_code}"
+}
+trap 'handle_error $LINENO' ERR
+
+die() {
+    local msg="$*"
+    echo >&2 "ERROR: $msg"
+    exit 1
+}
+
+# Check if daemon.json exists and parse configuration
+DAEMON_CONFIG_PATH="/run/peerpod/daemon.json"
+SCRATCH_ENABLED="false"
+ENCRYPTION_ENABLED="false"
+
+if [ -f "$DAEMON_CONFIG_PATH" ]; then
+    if command -v jq &>/dev/null; then
+        SCRATCH_ENABLED=$(jq -r '.["enable-scratch-disk"] // false' "$DAEMON_CONFIG_PATH")
+        ENCRYPTION_ENABLED=$(jq -r '.["enable-scratch-encryption"] // false' "$DAEMON_CONFIG_PATH")
+    else
+        # Fallback to grep if jq is not available
+        if grep -q '"enable-scratch-disk":.*true' "$DAEMON_CONFIG_PATH"; then
+            SCRATCH_ENABLED="true"
+        fi
+        if grep -q '"enable-scratch-encryption":.*true' "$DAEMON_CONFIG_PATH"; then
+            ENCRYPTION_ENABLED="true"
+        fi
+    fi
+fi
+
+echo "Scratch disk enabled: $SCRATCH_ENABLED"
+echo "Scratch encryption enabled: $ENCRYPTION_ENABLED"
+
+# Exit early if scratch disk is not enabled
+if [ "$SCRATCH_ENABLED" != "true" ]; then
+    echo "Scratch disk is disabled. Exiting."
+    exit 0
+fi
+
+# Check if the trusted_store device exists
+DISK_LABEL="trusted_store"
+DEV_PATH="/dev/disk/by-label/$DISK_LABEL"
+
+if [ ! -L "$DEV_PATH" ]; then
+    echo "Device for label $DISK_LABEL not found. Exiting."
+    exit 0
+fi
+
+# Resolve the device path
+DEV_PATH=$(readlink -f "$DEV_PATH")
+if [ ! -b "$DEV_PATH" ]; then
+    echo "Device $DEV_PATH is not a block device. Exiting."
+    exit 1
+fi
+
+# Create mount point
+MOUNT_POINT="/run/kata-containers/image"
+mkdir -p "$MOUNT_POINT"
+
+# Get device number for the encryption script
+DEV_NUM=$(lsblk -no MAJ:MIN "$DEV_PATH" | xargs)
+
+if [ "$ENCRYPTION_ENABLED" = "true" ]; then
+    # Generate encryption key and use LUKS encryption
+    echo "Setting up LUKS encrypted scratch storage"
+    head -c 4096 /dev/urandom >/run/encrypt_storage.key
+    /usr/local/bin/luks-encrypt-storage "$DEV_NUM" false "$MOUNT_POINT" /run/encrypt_storage.key
```

Comment:
we can maybe use a process substitution for the key file, then we also do not need to delete the key file in the luks script.

```suggestion
    /usr/local/bin/luks-encrypt-storage "$DEV_NUM" false "$MOUNT_POINT" <(head -c 4096 /dev/urandom)
```

Reply:
The luks script is from the kata and I kept it as is without any changes.

Reply:
Yeah, will change the core functionality of luks-encrypt-storage separately and try to consolidate between baremetal and peer-pods. For now I have kept the input handling as-is.